package.json

```
@@ -41,37 +41,37 @@
   "peerDependencies": {},
   "optionalDependencies": {},
   "dependencies": {
-    "class-validator": "0.13.1",
-    "express": "4.17.1",
+    "class-validator": "0.14.0",
+    "express": "4.18.2",
     "lodash": "4.17.21",
-    "nestjs-typeorm-paginate": "2.2.6",
-    "typescript": "4.4.4"
+    "nestjs-typeorm-paginate": "4.0.3",
+    "typescript": "4.9.4"
   },
   "peerDependency": {
-    "typeorm": "^0.2.30",
-    "@nestjs/common": "^8.1.2"
+    "typeorm": "^0.3.11",
+    "@nestjs/common": "^9.2.1"
```

can we maybe set this to
```suggestion
    "@nestjs/common": "^7.0.0 || ^8.0.0 || ^9.0.0"
```
unless there are known reasons for peer-linking this library to the latest NestJS?

---

But then shouldn't we do the same with TypeORM? 
i.e Nest v7 wouldn't be compatible with 0.3.*

Am I missing something here?

---

no, you aren't missing anything - I outright disconnected my brain and only looked at the `@nestjs/common` peer deps as I had this in mind for a while already and didn't consider the typeorm peer-dep 😓 

And yes, this is tricky, especially as `typeorm` is pre-1.0 and so according to semver "anything may change at any time". The list of breaking changes between 0.2 and 0.3 in the TypeORM changelog is scary. I guess that keeping the peer dep to the `0.3.x` minor version should be the wise thing to do here, even if Marxan Cloud will be sad, but that's another problem for another day.

---

Indeed the amount of breaking changes is scary, I actually had to refactor LG from removed / deprecated stuff. However, as far as I know, it does not bring any breaking changes to this library.

I guess we could also support multiple versions of typeorm, or even remove it from peer dependencies altogether, given the fact as you pointed out that we are in 0.* and anything can happen

---

I have no strong opinions (and surely not enough experience) on this, unfortunately. Given that our own projects are likely the only users of this library, I would go with what makes our lives simpler 😬 - feel free to decide 😉